### PR TITLE
Update/support type with uiLib

### PR DESCRIPTION
**Pull request by sirius651**
**Pull Request Summary**

> Support type augment on ui lib

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

스크린샷 2022-12-07 오후 11 10 57

스크린샷 2022-12-07 오후 11 11 37

### CODE REVIEW

File: src/@types/ui-lib.d.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import {
```
Comment (impelcrypto) [inline src/@types/ui-lib.d.ts:1]: Do we have to add this file in the portal? Is it possible to install @type dependency (like `@types/astar-ui`) under devDependencies?

**Comment by sirius651** [comment]
Sent PR to DefinitelyTyped [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63813)

**Comment by sirius651** [comment]
Just wanted to clean up this task. as discussed previous meeting, It was decided that we do not support DefinitelyTyped library to managing with individual modules because of complexity. Instead, it was modified to import the type from the ui library, not from the portal. please check. @impelcrypto @bobo-k2 @hoonsubin

**Review by impelcrypto — APPROVED** [review]
LGTM

My VS-code doesn't display the props information when I hover the mouse on [<astar-default-modal](https://github.com/AstarNetwork/astar-apps/blob/update/support-type-uilib/src/components/common/ModalWrapper.vue#L4) like this ↓.  

image

How about you @bobo-k2 @sirius651 ?

**Comment by sirius651** [comment]
> LGTM
> 
> My VS-code doesn't display the props information when I hover the mouse on [<astar-default-modal](https://github.com/AstarNetwork/astar-apps/blob/update/support-type-uilib/src/components/common/ModalWrapper.vue#L4) like this ↓.
> 
> image
> 
> How about you @bobo-k2 @sirius651 ?

I can reproduce this issue when clean up library, so I set exact path for exporting types. I believe it works. please check again @impelcrypto @bobo-k2

**Comment by impelcrypto** [comment]
hmm... still doesn't work on my end. How about you @bobo-k2 ?

image

**Comment by bobo-k2** [comment]
> hmm... still doesn't work on my end. How about you @bobo-k2 ?
Mine is like below, but it didn't to start immediately agter I updated dependencies. I needed to restart VS code.
image

**Comment by sirius651** [comment]
yeah, generally restarting is the key :)

**Comment by sirius651** [comment]
@bobo-k2 could you approve this PR if no issue?

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**